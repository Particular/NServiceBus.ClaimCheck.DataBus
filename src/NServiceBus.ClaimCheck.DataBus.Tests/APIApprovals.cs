namespace NServiceBus.DataBus.Tests;

using Features;
using NUnit.Framework;
using Particular.Approvals;
using PublicApiGenerator;

[TestFixture]
public class APIApprovals
{
    [Test]
    public void ApproveDataBus()
    {
        var publicApi = typeof(DataBusFeature).Assembly.GeneratePublicApi(new ApiGeneratorOptions
        {
            ExcludeAttributes = ["System.Runtime.Versioning.TargetFrameworkAttribute", "System.Reflection.AssemblyMetadataAttribute"]
        });
        Approver.Verify(publicApi);
    }
}