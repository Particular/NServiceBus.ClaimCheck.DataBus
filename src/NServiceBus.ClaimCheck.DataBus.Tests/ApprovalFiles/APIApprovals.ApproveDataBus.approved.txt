[assembly: System.Runtime.CompilerServices.InternalsVisibleTo(@"NServiceBus.ClaimCheck.DataBus.Tests, PublicKey=00240000048000009400000006020000002400005253413100040000010001007f16e21368ff041183fab592d9e8ed37e7be355e93323147a1d29983d6e591b04282e4da0c9e18bd901e112c0033925eb7d7872c2f1706655891c5c9d57297994f707d16ee9a8f40d978f064ee1ffc73c0db3f4712691b23bf596f75130f4ec978cf78757ec034625a5f27e6bb50c618931ea49f6f628fd74271c32959efb1c5")]
namespace NServiceBus.ClaimCheck.DataBus
{
    public static class ConfigureFileShareDataBus
    {
        public static NServiceBus.ClaimCheck.DataBus.DataBusExtensions<NServiceBus.ClaimCheck.DataBus.FileShareDataBus> BasePath(this NServiceBus.ClaimCheck.DataBus.DataBusExtensions<NServiceBus.ClaimCheck.DataBus.FileShareDataBus> config, string basePath) { }
    }
    public static class ConventionsBuilderExtensions
    {
        public static NServiceBus.ConventionsBuilder DefiningDataBusPropertiesAs(this NServiceBus.ConventionsBuilder builder, System.Func<System.Reflection.PropertyInfo, bool> definesDataBusProperty) { }
    }
    public class DataBusConventions
    {
        public DataBusConventions() { }
        public bool IsDataBusProperty(System.Reflection.PropertyInfo property) { }
    }
    public abstract class DataBusDefinition
    {
        protected DataBusDefinition() { }
        protected abstract System.Type ProvidedByFeature();
    }
    public class DataBusExtensions : NServiceBus.Configuration.AdvancedExtensibility.ExposeSettings
    {
        public DataBusExtensions(NServiceBus.Settings.SettingsHolder settings) { }
        public NServiceBus.ClaimCheck.DataBus.DataBusExtensions AddDeserializer<TSerializer>()
            where TSerializer : NServiceBus.ClaimCheck.DataBus.IDataBusSerializer, new () { }
        public NServiceBus.ClaimCheck.DataBus.DataBusExtensions AddDeserializer<TSerializer>(TSerializer serializer)
            where TSerializer : NServiceBus.ClaimCheck.DataBus.IDataBusSerializer { }
    }
    public class DataBusExtensions<T> : NServiceBus.ClaimCheck.DataBus.DataBusExtensions
        where T : NServiceBus.ClaimCheck.DataBus.DataBusDefinition
    {
        public DataBusExtensions(NServiceBus.Settings.SettingsHolder settings) { }
    }
    public class DataBusProperty<T> : NServiceBus.ClaimCheck.DataBus.IDataBusProperty
        where T :  class
    {
        public DataBusProperty() { }
        public DataBusProperty(T value) { }
        public bool HasValue { get; set; }
        public string Key { get; set; }
        [System.Text.Json.Serialization.JsonIgnore]
        public System.Type Type { get; }
        [System.Text.Json.Serialization.JsonIgnore]
        [System.Xml.Serialization.XmlIgnore]
        public T Value { get; }
        public object GetValue() { }
        public void SetValue(object valueToSet) { }
    }
    public class FileShareDataBus : NServiceBus.ClaimCheck.DataBus.DataBusDefinition
    {
        public FileShareDataBus() { }
        protected override System.Type ProvidedByFeature() { }
    }
    public interface IDataBus
    {
        System.Threading.Tasks.Task<System.IO.Stream> Get(string key, System.Threading.CancellationToken cancellationToken = default);
        System.Threading.Tasks.Task<string> Put(System.IO.Stream stream, System.TimeSpan timeToBeReceived, System.Threading.CancellationToken cancellationToken = default);
        System.Threading.Tasks.Task Start(System.Threading.CancellationToken cancellationToken = default);
    }
    public interface IDataBusProperty
    {
        bool HasValue { get; set; }
        string Key { get; set; }
        System.Type Type { get; }
        object GetValue();
        void SetValue(object value);
    }
    public interface IDataBusSerializer
    {
        string ContentType { get; }
        object Deserialize(System.Type propertyType, System.IO.Stream stream);
        void Serialize(object databusProperty, System.IO.Stream stream);
    }
    public class SystemJsonDataBusSerializer : NServiceBus.ClaimCheck.DataBus.IDataBusSerializer
    {
        public SystemJsonDataBusSerializer() { }
        public string ContentType { get; }
        public object Deserialize(System.Type propertyType, System.IO.Stream stream) { }
        public void Serialize(object dataBusProperty, System.IO.Stream stream) { }
    }
    public static class UseDataBusExtensions
    {
        public static NServiceBus.ClaimCheck.DataBus.DataBusExtensions UseDataBus(this NServiceBus.EndpointConfiguration config, System.Func<System.IServiceProvider, NServiceBus.ClaimCheck.DataBus.IDataBus> dataBusFactory, NServiceBus.ClaimCheck.DataBus.IDataBusSerializer dataBusSerializer) { }
        public static NServiceBus.ClaimCheck.DataBus.DataBusExtensions<TDataBusDefinition> UseDataBus<TDataBusDefinition>(this NServiceBus.EndpointConfiguration config, NServiceBus.ClaimCheck.DataBus.IDataBusSerializer dataBusSerializer)
            where TDataBusDefinition : NServiceBus.ClaimCheck.DataBus.DataBusDefinition, new () { }
        public static NServiceBus.ClaimCheck.DataBus.DataBusExtensions<TDataBusDefinition> UseDataBus<TDataBusDefinition, TDataBusSerializer>(this NServiceBus.EndpointConfiguration config)
            where TDataBusDefinition : NServiceBus.ClaimCheck.DataBus.DataBusDefinition, new ()
            where TDataBusSerializer : NServiceBus.ClaimCheck.DataBus.IDataBusSerializer, new () { }
    }
}
namespace NServiceBus.Features
{
    public class DataBusFeature : NServiceBus.Features.Feature
    {
        protected override void Setup(NServiceBus.Features.FeatureConfigurationContext context) { }
    }
}